### PR TITLE
First set for the support of OS: Debian 9,10 and Ubuntu 18.04 and 20.04 

### PR DESCRIPTION
**Pull request by vibhorkumar123**
This is still a work in progress. With this commit, the following roles support Ubuntu (18.04,20.04) and Debian (9,10).
1. setup_repo
2. install_dbserver
3. init_dbserver
4. setup_replication
5. setup_efm
6. setup_pemserver
7. setup_pemagent
8. manage_dbserver

We have also changed the following variable name, which will impact the playbook.yml and ansible command:
1. yum_username -> repo_username
2. yum_password -> repo_password.

### CODE REVIEW

**Comment by vibhorkumar123** [comment]
@jt-edb This completes the first set of roles, which has the support of Ubuntu and Debian. Can you please review this PR?

File: roles/setup_pemagent/tasks/pem_agent_install.yml [hunk 1]
```diff
@@ -3,12 +3,20 @@
   yum:
     name: "{{ pem_agent_package }}"
     state: present
-  when: os in [ 'CentOS7','RHEL7']
+  when: os in [ 'CentOS7','RedHat']
```
Comment (jt-edb) [inline roles/setup_pemagent/tasks/pem_agent_install.yml:6]: `RedHat7`

File: roles/setup_pemserver/defaults/main.yml [hunk 2]
```diff
@@ -19,6 +19,8 @@ pem_agent_service: pemagent
 
 pem_server_bin_path: "/usr/edb/pem/bin"
 pem_agent_bin_path: "/usr/edb/pem/agent/bin"
+pem_server_config_file: "{{ '/etc/httpd/conf.d/edb-ssl-pem.conf' if ansible_os_family == 'RedHat' else '/etc/apache2/sites-available/edb-ssl-pem.conf' }}"
+pem_server_port: "8443"
```
Comment (jt-edb) [inline roles/setup_pemserver/defaults/main.yml:23]: What's going on if we change it to `8444`? Is the Apache configuration updated accordingly?

File: roles/setup_repo/tasks/PG_RedHat_setuprepos.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  dnf:
+    name:
+      - "{{ epel_repo_8 }}"
+    state: present
+  when:
+    - ansible_distribution_major_version == "8"
+  become: yes
+
+- name: Install PG repo for RedHat 8
+  yum:
```
Comment (jt-edb) [inline roles/setup_repo/tasks/PG_RedHat_setuprepos.yml:54]: We should use the `dnf` module in that case.

File: roles/setup_repo/tasks/setup_repo.yml [hunk 4]
```diff
@@ -36,14 +39,23 @@
 
 - name: Cleanup the repos based on force_install
   block:
-    - include_tasks: "{{ pg_type }}_rhel_rm_repos.yml"
+    - include_tasks: "PG_RedHat_rm_repos.yml"
```
Comment (jt-edb) [inline roles/setup_repo/tasks/setup_repo.yml:42]: You should add a condition to restrict this task for RedHat based OS

File: roles/setup_pemagent/tasks/pem_agent_install.yml (hunk 1 above)
Comment (vibhorkumar123) [reply roles/setup_pemagent/tasks/pem_agent_install.yml]: Fixed!

File: roles/setup_pemserver/defaults/main.yml (hunk 2 above)
Comment (vibhorkumar123) [reply roles/setup_pemserver/defaults/main.yml:23]: I think this should allow the changes
https://github.com/EnterpriseDB/edb-ansible/blob/8901e10c34926578b68053f51c5b5721980837ca/roles/setup_pemserver/tasks/pem_server_config.yml#L25

File: roles/setup_repo/tasks/PG_RedHat_setuprepos.yml (hunk 3 above)
Comment (vibhorkumar123) [reply roles/setup_repo/tasks/PG_RedHat_setuprepos.yml]: Fixed. I made some more changes in PG_RedHat_setuprepos.yml. Can you please review the new change?

File: roles/setup_repo/tasks/setup_repo.yml (hunk 4 above)
Comment (vibhorkumar123) [reply roles/setup_repo/tasks/setup_repo.yml]: FIxed. Changed it to something like:
`PG_{{ ansible_os_family }}_rm_repos`

File: roles/setup_pemserver/defaults/main.yml (hunk 2 above)
Comment (jt-edb) [reply roles/setup_pemserver/defaults/main.yml:23]: I missed it, sorry.

File: roles/setup_repo/tasks/PG_RedHat_setuprepos.yml (hunk 3 above)
Comment (jt-edb) [reply roles/setup_repo/tasks/PG_RedHat_setuprepos.yml]: Sounds good.

**Review by jt-edb — APPROVED** [review]

**Pull request closed without merging**